client-side/src/graphs/graphs.jsx

```diff
@@ -0,0 +1,313 @@
+import * as React from 'react';
+import { LineChart, ChartContainer, LinePlot, BarPlot, ChartsYAxis, ChartsXAxis, Gauge, PieChart, ChartsLegend } from '@mui/x-charts';
+import Stack from '@mui/material/Stack';
+import { useQuery, gql } from '@apollo/client';
+
+const GET_VISITED_WEBSITES = gql`
+query GetVisitedWebsites {
+visitedWebsites {
+ websiteId {
+   name
+   url
+ }
+  visitsTime {
+    activityTime
+    visitDate
+  }
+}}
+`;
+
+const GET_WEBSITE = gql`
+query website{
+  websites {
+    name
+    url
+  }
+}
+`
+const GET_USERS = gql`
+ query users{
+   users {
+     name
+     password
+     email
+    profileImage
+    visitsWebsites {
+      visitsTime {
+        activityTime
+        visitDate
+      }
+      websiteId {
+        name
+        url
+      }
+    }
+   }
+ }`
+
+
+const otherProps = {
+    width: 400,
+    height: 200,
+    sx: {
+        [`.${ChartsLegend.root}`]: {
+            transform: 'translate(20px, 0)',
+        },
+    },
+};
+
+
+const VisitedWebsitesComponent = ({ startDate, endDate, user }) => {
+    const websites = useQuery(GET_WEBSITE);
+    const visitedWebsite = useQuery(GET_VISITED_WEBSITES);
+    const users = useQuery(GET_USERS);
+    const colors = ["red", "yellow", "orange", "blue", "lightgreen"]
+    let dateStart = new Date(startDate.$d);
+    let year = dateStart.getFullYear();
+    let month = ('0' + (dateStart.getMonth() + 1)).slice(-2);
+    let day = ('0' + dateStart.getDate()).slice(-2);
+    dateStart = `${year}-${month}-${day}`
+    let dateEnd = new Date(endDate.$d);
+    year = dateEnd.getFullYear();
+    month = ('0' + (dateEnd.getMonth() + 1)).slice(-2);
+    day = ('0' + dateEnd.getDate()).slice(-2);
+    dateEnd = `${year}-${month}-${day}`
+    if (websites.loading || users.loading) return <p>Loading...</p>;
+
+    if (websites.error  || users.error) return <p>Error</p>;
+
+    let webSite = websites.data.websites.map((obj, index) => ({ ...obj, color: colors[index] }))
+    const u = users.data.users.find(u => u.email === user.email)
+    console.log({ u });
+    
+    const totalActivityTimeByWebsite = {};
+
+    u.visitsWebsites.forEach(visitedWebsite => {
+        const websiteName = visitedWebsite.websiteId.name;
+        const activityTime = visitedWebsite.visitsTime.reduce((total, visit) => {
+            let date = new Date(visit.visitDate)
+            year = date.getFullYear();
+            month = ('0' + (date.getMonth() + 1)).slice(-2);
+            day = ('0' + date.getDate()).slice(-2);
+            date = `${year}-${month}-${day}`
```

Comment:
please create a function for formatting date and use it in all places that you creating a date.